app/controllers/users/sessions_controller.rb

```diff
@@ -0,0 +1,11 @@
+module Users
+  class SessionsController < Devise::SessionsController
+   def new
+     if  request.referer.present?
+     	self.resource = resource_class.create(sign_in_params)
+        store_location_for(resource, request.referer)  
```

Comment:
Let's only store the referrer if it under the same domain that the application is running under. We don't want to redirect people to external sites.